java/src/main/java/ai/onnxruntime/OrtProviderOptions.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package ai.onnxruntime;
+
+import java.io.IOException;
+
+/** An abstract base class for execution provider options classes. */
+// Note this lives in ai.onnxruntime to allow subclasses to access the OnnxRuntime.ortApiHandle
+// package private field.
+public abstract class OrtProviderOptions implements AutoCloseable {
+  static {
+    try {
+      OnnxRuntime.init();
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to load onnx-runtime library", e);
+    }
+  }
+
+  protected final long nativeHandle;
+
+  /**
+   * Constructs a OrtProviderOptions wrapped around a native pointer.
+   *
+   * @param nativeHandle The native pointer.
+   */
+  protected OrtProviderOptions(long nativeHandle) {
+    this.nativeHandle = nativeHandle;
+  }
+
+  /**
+   * Allow access to the api handle pointer for subclasses.
+   *
+   * @return The api handle.
+   */
+  protected static long getApiHandle() {
+    return OnnxRuntime.ortApiHandle;
+  }
+
+  /**
+   * Gets the provider enum for this options instance.
+   *
+   * @return The provider enum.
+   */
+  public abstract OrtProvider getProvider();
+
+  @Override
+  public void close() {
+    close(OnnxRuntime.ortApiHandle, nativeHandle);
+  }
+
+  /**
+   * Native close method.
+   *
+   * @param apiHandle The api pointer.
+   * @param nativePointer The native options pointer.
+   */
+  protected abstract void close(long apiHandle, long nativePointer);
```

Comment:
>nativePointer [](http://example.com/codeflow?start=53&length=13)

What is the difference between nativeHandle and nativePointer?

Reply:
I mean the same thing by `nativePointer` and `nativeHandle`, so that should be `nativeHandle` for consistency.